app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/RtcServerConnectionState.kt

```diff
@@ -0,0 +1,36 @@
+package co.netguru.baby.monitor.client.feature.server
+
+import co.netguru.baby.monitor.client.feature.communication.webrtc.observers.DefaultObserver
+import io.reactivex.Observable
+import io.reactivex.subjects.PublishSubject
+import org.webrtc.PeerConnection
+
+class RtcServerConnectionObserver : DefaultObserver() {
+    private val publishSubject = PublishSubject.create<RtcServerConnectionState>()
+    val rtcConnectionObservable: Observable<RtcServerConnectionState> = publishSubject
+
+    fun onAcceptOffer() {
+        publishSubject.onNext(RtcServerConnectionState.ConnectionOffer)
+    }
+
+    override fun onIceConnectionChange(iceConnectionState: PeerConnection.IceConnectionState?) {
+        super.onIceConnectionChange(iceConnectionState)
+        when (iceConnectionState) {
+            PeerConnection.IceConnectionState.CONNECTED -> publishSubject.onNext(
+                RtcServerConnectionState.Connected
+            )
+            PeerConnection.IceConnectionState.DISCONNECTED -> publishSubject.onNext(
+                RtcServerConnectionState.Disconnected
+            )
+            PeerConnection.IceConnectionState.CLOSED -> publishSubject.onComplete()
+            else -> Unit
+        }
+    }
+}
+
+sealed class RtcServerConnectionState {
+    object ConnectionOffer : RtcServerConnectionState()
+    object Connected : RtcServerConnectionState()
+    object Disconnected : RtcServerConnectionState()
+    object Error : RtcServerConnectionState()
+}
```

Comment:
I don't think modeling state this way is a good approach. There was such thing in legacy codebase which caused me headache a couple of times.

Reply:
Any alternatives and why is it bad?